src/main/java/core/basesyntax/bookstore/repository/book/BookRepository.java

```diff
@@ -1,9 +1,25 @@
 package core.basesyntax.bookstore.repository.book;
 
 import core.basesyntax.bookstore.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {

```

Comment:
Do you really need to fetch categories for all methods? And if you do so, specify in method name that your method fetches categories too, similar to what you named it in `findAllWithCategories`